Automattic/jetpack
# Map block: fix script loading to work with FSE

### Description
Possible fix for: #19482

Currently the Jetpack Map block fails to load in the Site Editor context because the `mapbox-gl.js` file is loaded in the main window context, but the block editor is running in an iframe when in the site editor context.

To get around this issue this PR adds a supporting library which can load editor asset files into the correct window context of the current editor. 

This issue as been raised for discussion at https://github.com/WordPress/gutenberg/issues/31022 in order to see if there is a broader solution that might be possible within the core Gutenberg plugin.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Check out this PR and run in local env along with latest Gutenberg plugin and an FSE compatible theme
- Add a Map block in both a post and in the site editor and check that the map loads in the editor in both places, and markers can be added
- Check the maps display in the front end in post body and in site headers or footers

**Please note:** This is still a PoC solution and open for review on the general approach. Waiting for feedback on https://github.com/WordPress/gutenberg/issues/31022 before merging
